benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/services/external/FabanManagerService.java

```diff
@@ -153,32 +172,46 @@ public FabanStatus pollForTrialStatus(String trialID, RunId runId) throws RunIdN
     // B) wait/poll for trial to complete and store the trial result in the DB
     // TODO - is this the status we want to use? No it is a subset, should also
     // include metrics computation status
-    RunStatus status = fabanClient.status(runId);
+    RunStatus status;
 
-    // workaround for issue #409 (FabanClient throws an exception when status is UNKNOWN)
-    // wait 60s before polling (Faban needs time to setup)
     try {
-      Thread.sleep(60 * 1000);
-    } catch (InterruptedException e) {
-      e.printStackTrace();
-    }
-
-    while (status.getStatus().equals(QUEUED) || status.getStatus().equals(RECEIVED)
-        || status.getStatus().equals(STARTED)) {
 
+      // wait 60s before polling (Faban needs time to setup)
       try {
-        Thread.sleep(30 * 1000);
+        Thread.sleep(60 * 1000);
       } catch (InterruptedException e) {
         e.printStackTrace();
       }
 
       status = fabanClient.status(runId);
 
-    }
+      while (status.getStatus().equals(QUEUED) || status.getStatus().equals(RECEIVED)
+          || status.getStatus().equals(STARTED) || status.getStatus().equals(UNKNOWN)) {
 
-    RunInfo runInfo = fabanClient.runInfo(runId);
+        try {
+          Thread.sleep(30 * 1000);
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+
+        status = fabanClient.status(runId);
```

Comment:
You can end up here also if you have an exception, and that should not be the case.